liwords-ui/src/lobby/seek_form.tsx

```diff
@@ -224,21 +233,29 @@ export const SeekForm = (props: Props) => {
     setTtag(tt);
   };
 
-  const onUsernameSearch = (searchText: string) => {
-    axios
-      .post<SearchResponse>(
-        toAPIUrl('user_service.AutocompleteService', 'GetCompletion'),
-        {
-          prefix: searchText,
-        }
-      )
-      .then((resp) => {
-        console.log('resp', resp.data);
-        setUsernameOptions(
-          !searchText ? [] : resp.data.users.map((u) => u.username)
-        );
-      });
-  };
+  const onUsernameSearch = useCallback(
+    (searchText: string) => {
+      axios
+        .post<SearchResponse>(
+          toAPIUrl('user_service.AutocompleteService', 'GetCompletion'),
+          {
+            prefix: searchText,
+          }
+        )
+        .then((resp) => {
+          console.log('resp', resp.data);
+          const defaultOptions = tournamentID
+            ? presences.map((p) => p.username).filter((u) => u !== username)
+            : Object.values(friends).map((f) => f.username);
+          setUsernameOptions(
+            !searchText
+              ? defaultOptions || []
```

Comment:
(performance) `defaultOptions` is reevaluated on every `onUsernameSearch` (every keystroke), even when it isn't used. it should just be evaluated if `!searchText`.